pkg/controllers/management/drivers/kontainerdriver/kontainerdriver.go

```diff
@@ -349,36 +327,69 @@ func (l *Lifecycle) Updated(obj *v3.KontainerDriver) (runtime.Object, error) {
 		return obj, nil
 	}
 
-	if obj.Spec.BuiltIn {
+	if obj.Spec.BuiltIn && v3.KontainerDriverConditionActive.IsTrue(obj) {
 		// Builtin drivers can still have their schema change during Rancher upgrades so we need to try
 		return obj, l.updateDynamicSchema(obj)
 	}
 
+	var err error
+	var tmpObj runtime.Object
+	newObj := obj.DeepCopy()
+	switch {
+	case !obj.Spec.Active: // dealing with deactivate action
+		tmpObj, err = v3.KontainerDriverConditionInactive.DoUntilTrue(newObj, func() (runtime.Object, error) {
+			if err = l.dynamicSchemas.Delete(getDynamicTypeName(newObj), &v13.DeleteOptions{}); err != nil && !errors.IsNotFound(err) {
+				return nil, fmt.Errorf("error deleting schema: %v", err)
+			}
+			if err = l.removeFieldFromCluster(newObj); err != nil {
+				return nil, err
+			}
+
+			RemoveCondition(newObj, v3.KontainerDriverConditionActive)
```

Comment:
Why do we need to remove the Active condition vs just setting it to false?

Reply:
We will translate condition to state in https://github.com/rancher/types/blob/master/status/status.go#L176. And 
- `Active = false` means something wrong with active
- `Active = Unknown` means transitioning
And I can not keep it true because I need that condition to drive `createDynamicSchema` action. So removing it is the only option here.

Reply:
Updated the logic and we don't need to remove condition anymore.